### PR TITLE
perf: switch to sharedarraybuffer to reduce serialization and messaging

### PR DESCRIPTION
**Pull request by sverben**

### CODE REVIEW

File: libserial/LibSerial.c [hunk 1]
```diff
@@ -36,36 +51,44 @@ EM_ASYNC_JS(void, clear_read_buffer, (int timeoutMs), {
...
+        result.set(window.readBuffer.slice(window.readAddress, targetAddress), read);
+        read += targetAddress - window.readAddress;
+
+        window.readAddress = targetAddress;
+        if (window.readAddress === window.readBuffer.length) {
+            window.readAddress = 3;
+        }
+    }
+
+    if (read === 0) {
```
Comment (rmoesbergen) [inline libserial/LibSerial.c:81]: This check can be improved: what if some data was read, and then we run into a timeout?

File: libserial/LibSerial.c [hunk 2]
```diff
@@ -18,9 +18,24 @@ void errorCallback() {
...
+        const initialLength = Math.min(len - written, window.writeBuffer.length - writeAddress);
+        const data = new Uint8Array(Module.HEAPU8.buffer.slice(buf + written, buf + written + initialLength));
+        window.writeBuffer.set(data, writeAddress);
+        window.avrdudeLog = [...window.avrdudeLog, "Sending: " + Array.from(data).join(",")];
+
+        written += initialLength;
+        writeAddress += initialLength;
+
+        if (writeAddress === window.writeBuffer.length) {
+            writeAddress = 3;
```
Comment (rmoesbergen) [inline libserial/LibSerial.c:34]: why 3? Could use a comment

**Review by rmoesbergen — APPROVED** [review]
🚀

**Comment by koen1711** [comment]
Will be reviewing shortly

**Comment by koen1711** [comment]
Linux :heavy_check_mark: 
Windows :heavy_check_mark: 
MacOS :clock1130:

**Review by koen1711 — APPROVED** [review]

**Pull request merged**